sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5770,6 +5905,46 @@ bool _pi_queue::useCopyEngine(bool PreferCopyEngine) const {
          (!isInOrderQueue() || UseCopyEngineForInOrderQueue);
 }
 
+// Wait on all operations in flight on this Queue.
+// If using immediate commandlists add barriers to all commandlists associated
+// with the Queue. An alternative approach would be to wait on all Events
+// associated with the in-flight operations. For standard commandlists sync the
```

Comment:
Are we collecting all events submitted to the immediate command-list currently? That can become huge for a long running app. I think we need to do "garbage collection" on the list, at least remove events whose only remaining reference is from this same list, i.e. SYCL RT already released its references.

Reply:
This is a potential problem. However, I think the SYCL runtime will need to release the events. In the plugin we don't have anything in the commandlist to track. It is not batched. We cannot tell if the application will wait on an event in the future.

Reply:
After SYCL RT calls piEventRelease we know it is not going to wait for it, or use it in any other way. At this time we should remove it from the list here. I think it can be done in a separate PR, but add TODO comment now.